lib/chef/resource/dsc_resource.rb

```diff
@@ -31,6 +31,7 @@ def initialize(name, run_context)
         super
         @properties = {}
         @resource = nil
+        @allow_reboot = :nothing

```

Comment:
weird things will happen if someone sets allow_reboot to true or false. I'd expect this property to take boolean and then have another method choose the appropriate action to pass to `@reboot_resource.run_action` in `invoke_reboot`.

althernatively I'd rename the property to reboot_action, but I think that's unnecessarily complicated.